src/core/ddsi/include/dds/ddsi/ddsi_typelookup.h

```diff
@@ -76,17 +77,17 @@ extern size_t typelookup_service_reply_nops;
 
 int ddsi_tl_meta_equal (const struct tl_meta *a, const struct tl_meta *b);
 uint32_t ddsi_tl_meta_hash (const struct tl_meta *tl_meta);
-void ddsi_tl_meta_ref (struct ddsi_domaingv *gv, const type_identifier_t *type_id, const struct ddsi_sertype *type, const ddsi_guid_t *src_ep_guid, const ddsi_guid_prefix_t *dst);
+void ddsi_tl_meta_ref (struct ddsi_domaingv *gv, const type_identifier_t *type_id, const struct ddsi_sertype *type, const ddsi_guid_t *proxy_ep_guid, const ddsi_guid_prefix_t *dst);
```

Comment:
I think it would be advisable to have different versions of `meta_ref` and `meta_unref` so that there are fewer null pointers being passed in. For `ref` it really splits into local entities vs proxies, and for `unref` based on type_id or type.

I am fine with these being wrappers that forward it to the current implementation, it's just that I think it'd be clearer at the call sites.

Reply:
For both `ref` and `unref` I've split up the function into a `proxy` and `local` variant; I think it's more clear to use proxy/local this for unref too, instead of having a `type_id` and `type` variant.